mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -716,14 +939,22 @@ int main(int argc, char **argv) {
     exit(1);
   }
 
+  // populate host harness and host validation.
+  if (populateValidation.getValue()) {
+    if (failed(populateHostHarnessLogic(module, builder, context, dataType))) {
+      llvm::errs() << "Host validation populated failed.\n";
+      exit(1);
+    }
+  }
+
   // Apply passes.
   if (failed(runMLIRPasses(module, passPipeline, kernelName))) {
     llvm::errs() << "Lowering failed.\n";
     exit(1);
   }
 
   // populate host launch logic.
-  if (useHostHarness.getValue()) {
+  if (useHostHarness.getValue() || populateValidation.getValue()) {
```

Comment:
Same as above. Decouple the validation and host harness routines otherwise this looks very confusing.